### ISSUE TITLE
oauth redirect uri not used when calling identity provider

### ISSUE THREAD

**Issue by Tynamix** [springdoc/springdoc-openapi#340]
Like i already saw in #320 , the html for oauth2 redirect is located here: http://server:host/swagger-ui/oauth2-redirect.html. This works!
But when trying to authenticate via oauth2 directly in the swagger ui to an identity provider, the redirect uri given to the identity provider will be http://server:host/oauth2-redirect.html (without the swagger-ui subpath).

**Comment by bnasslahsen** [comment]
Dis you declare?

`springdoc.swagger-ui.oauth2RedirectUrl=http://server:host/oauth2-redirect.html`

**Comment by Tynamix** [comment]
Hey @bnasslahsen !
Thanks for this project. I do now know the issue. It works perfect, when this framework generates the documentation. In the case (like me) i provide my own YAML file, the authorization doesnt work.

To reproduce this issue, follow this steps. Use this swagger file:

```
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Sample API
  description: NA
paths:
  /customers:
    post:
      tags:
        - Customer related operations
      description: Registers a new customer.
      operationId: registerCustomer
      requestBody:
        description: Customer to register
        required: true
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/NewCustomer'
      responses:
        '200':
          description: customer response
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Customer'

components:

  securitySchemes:
    oAuthSample:    # <---- arbitrary name
      type: oauth2
      description: You need a client id and client secret. Ask your support.
      flows:
        authorizationCode:   # <---- OAuth flow(authorizationCode, implicit, password or clientCredentials)
          authorizationUrl: http://someAuthURL
          tokenUrl: http://someTokenURL


  schemas:

    Customer:
      type: object
      properties:
        customerId:
          type: integer
          format: int64
        firstName:
          type: string
          maxLength: 100
        lastName:
          type: string
          maxLength: 100
        email:
          type: string
          maxLength: 100
        phone:
          type: string
          maxLength: 64
        newsletter:
          type: boolean
          default: false
        phoneContact:
          type: boolean
          default: false

    NewCustomer:
      type: object
      required:
        - email
        - password
        - salutation
        - firstName
        - lastName
        - language
      properties:
        email:
          type: string
          maxLength: 100
          format: email
        password:
          type: string
          maxLength: 64
        firstName:
          type: string
          maxLength: 100
        lastName:
          type: string
          maxLength: 100
        phone:
          type: string
          maxLength: 64
        newsletter:
          type: boolean
          default: false
        phoneContact:
          type: boolean
          default: false
        affiliateId:
          type: string
          maxLength: 64
```

use your  framework and call this site: /swagger-ui/index.html?url=PATHTOCUSTOM.yaml

Click on authorize and enter some data for client id and client secret. click on authorize --> wrong redirect uri.

Maybe this get fixed when releasing the bugfix of #339 ?

**Comment by bnasslahsen** [comment]
On the next release, the issue #339 will be integrated.
Maybe you are serving a static OpenAPI description, and not using the generated OpenAPI spec. Right ?
